monkey/infection_monkey/config.py

```diff
@@ -135,6 +138,8 @@ def as_dict(self):
 
     # depth of propagation
     depth = 2
+    max_depth = None
+    started_on_island = False
```

Comment:
I'm really uncomfortable with this GLOBAL configuration parameter being a part of every monkey's config. Perhaps this is the catalyst for generating a global conf and a single monkey conf. Thoughts?

Reply:
We already have global and per monkey conf (Yeah, it was a discovery for me also). ATM the codebase has no indication of these two separate configs, they are mushed into same services etc. To refactor config in general would be a good idea, but if we can let's not mix performance and tech-debt together. Better figure out how much time we want to set for eliminating tech debt and then work on it separatelly